### DISCUSSION TITLE
Can't finish downloading file, side downloading problem

### DISCUSSION THREAD

**Discussion by Avel757** [aclap-dev/video-downloadhelper#99] [Help and Q&A]
### Your question

Just start downloading HLS file and it doesnt let me finish.
I use vpn of kazachstan and trying to download series from kazach web site with subscription. Bought subscription but cant download through app even after buying license. 

https://tv.telecom.kz/vod/salem/1286serial/e_1286serial-s1-e2_1/watch  

Code of error.

[tls @ 000002545cf53880] IO error: End of file
[https @ 000002545d02ee40] Stream ends prematurely at 736906, should be 4909432
[tls @ 000002545cf53880] IO error: End of file
[tls @ 000002545cf4bdc0] IO error: End of file
[https @ 000002545cf4c700] Stream ends prematurely at 491146, should be 4521024
[tls @ 000002545cf4bdc0] IO error: End of file
[tls @ 000002545d49f8c0] IO error: End of file
[https @ 000002545dcd3a00] Stream ends prematurely at 376458, should be 4862056
[tls @ 000002545d49f8c0] IO error: End of file
    Last message repeated 1 times
[https @ 000002545dcd3a00] Stream ends prematurely at 376458, should be 4877660
[tls @ 000002545d49f8c0] IO error: End of file
[aac_adtstoasc @ 000002545d023280] Error parsing ADTS frame header!
[mp4 @ 000002545d56ecc0] Error applying bitstream filters to an output packet for stream #1: Invalid data found when processing input
[aost#0:1/copy @ 000002545d399640] Error submitting a packet to the muxer: Invalid data found when processing input
[out#0/mp4 @ 000002545dd189c0] Error muxing a packet

### Addon & CoApp version and details

Windows x64
ver 8.2.0.23 
app companion VdhCoApp 2.0.17
license

**Comment by paulrouget** [comment]
If fails after how long? 10%? 90%?

**Reply by Avel757** [reply]
till 10, sometimes a bit more, once even was till 85%

**Reply by paulrouget** [reply]
Understood. Thank you. I'll look it up.

**Reply by paulrouget** [reply]
Can you take a screenshot mid download please? Need to understand which download method you're using.

**Reply by Avel757** [reply]
![Снимок экрана 2024-02-19 140432](https://github.com/aclap-dev/video-downloadhelper/assets/100200112/3246febd-d6bc-44e4-898d-8042795b822f)

**Reply by Avel757** [reply]
Are there any news about my problem?

**Reply by GrampaWildWilly** [reply]
I don't work for VDH.  I'm just another user.  I tried to visit that site.  It's all in Russian, as far as I can tell.  Google Translate didn't do a very good job with the page.  There's a popup that appears on the untranslated page that does not appear on the translated page so I have no idea what it says.  There's 2 buttons on it but I'm not clicking buttons I can't read.  I think it's the usual nag about accepting or rejecting cookies.  But that's a guess & I'm not going on guesses on this site.  I did try to click what appeared to be a link to a movie, but when I went there & clicked what looked like a play button, it took me to a screen that looked a lot like a login page.  So I am going to guess that this site requires a registration.  I am not willing to sign up for something I can't even get properly translated.  I suspect that will be the reaction you'll get from the VDH folks.  I suspect.  I'm guessing again.  I don't speak for them.  They are quite capable of speaking for themselves.  But I don't think you're going to get any meaningful help here, partly because of problems with translation, partly because the site requires a registration.  If you can post a page that Google Translate will translate well & that does not require a login (even if it is free), then maybe you'll get some help.  Otherwise, I believe you are out of luck here.

**Reply by GrampaWildWilly** [reply]
Also, properly fill in the section that asks for your add-on & CoApp version & details.  Follow instructions you can find by clicking the Wiki link at the top of this page.  On that Wiki page, look for a link that says, "I need help."  Click that.  Then do what it says.  Don't abbreviate.  Don't vary.  Don't omit.  Do what it says.  You might also consider some other links on the Wiki page.  I'm thinking specifically of the link to the latest beta versions & the link to the latest CoApp version.  You really should get on the latest releases of everything before you report a problem.  Bugs have been corrected in newer versions.  If you're not running these latest versions, you may be encountering problems that are solved in the latest releases.  So get on the latest releases & see if that helps at all.

**Reply by GrampaWildWilly** [reply]
Here's what I'm talking about.  Clicking the link you provided gives me this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f8b0f40c-f109-4875-adb8-5d21308cfb5a)

Google Translate gives me this:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/de8fe4f7-d336-4610-9eeb-b3a1c1aad84c)

Useless.  Clicking the first thumbnail on the untranslated page gives me this:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/ab37c9e1-3a93-4584-8f96-f8914750e636)

Clicking that other button on the bottom right gives me this:

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/7e815702-3ea3-4f4c-ba7b-4d8eff37d0e8)

This is a dead end for me.  I certainly can't help you if this is all I can get.  No, I will not allow you to share your login credentials with me.  The pages don't translate properly.  I'm not going to help any further here.

**Reply by Avel757** [reply]
Hey there, thanks for reply. Look, this is a web site that requires account and subscription, I be made both of them and already have access to series and video himself. If you need screenshots or something specific from video player in browser, tell me and I will be able to navigate you. I understand that it’s hard without translation, but I appreciate any help and ready to help understanding this.

**Reply by GrampaWildWilly** [reply]
Oh no.  You won't be navigating me.  You will be navigating.  But showing us some screenshots is EXACTLY what would be needed here.  And I repeat: I don't work for VDH.  I'm just another user.

First, make sure you are using the latest versions of the software.  You still haven't followed the instructions for telling us what your environment is.  Go to the Wiki & read the section about "I need help."  Every word of it.  Then follow those instructions.  There are links within the instructions.  Click the links & read what you find.  We need to know all of that information.

Also in the Wiki you will see links to get the public versions of VDH & the beta versions.  If there is a beta version that is newer than the public version, you want to be using the beta.  How do you figure that out?  I don't know.  You haven't f0llowed the instructions for posting a new problem report.  I don't know what you're running.

There is also a link in the Wiki to the CoApp.  Get the latest version of that for your operating system.

Now for your screenshots . . .

Don't start with your login.  Start on the web page of a movie.  Show us the VDH menu you get when you first visit a movie page, before you even click play.  Let's hope VDH has recognized something.  If the menu is so long it's got a scroll bar, take multiple screenshots to show the whole thing.  If the VDH menu is empty, then you will need to click play, then pause playback after a couple of seconds.  That should get VDH to recognize something if it hadn't before.  We'll want to see that VDH menu.

Then indicate which variant you are going to download & show us (screenshots) the steps of the file selection dialog.

Once the download is launched, click on the blue dot in the VDH menu & show us (screenshot) the download status menu.  Wait until a few percent, maybe 20% of the file has been downloaded so we can get an idea of what sort of download speed you're getting.

Then show us in a screenshot whatever result you get.  Maybe this is the VDH menu again.  Maybe this is a screenshot of your file manager showing the downloaded file.  The file probably has properties.  Show us screenshots of the properties.  Maybe you got an error.  Show a screenshot.  If there are error details, that is just text.  Paste the text into your post.

Be sure you translate things for us so we can understand what you are getting.

It's a bit more work when we're dealing with something behind a logon.  But it can be done.  You have to pretend we're watching over your shoulder & you're pointing things out to us.  You are our eyes when you post screenshots.  Let's see if we can make some progress here.

**Reply by GrampaWildWilly** [reply]
Here's a sample of the kind of thing I'm expecting from you.  You will probably have some different results since you are having problems.  But this should give you an idea of the kind of screenshots I'm expecting from you.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f66886f7-833e-4087-9dba-a2e970f789f9)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/8847446a-3734-425f-8446-ed825d9e5ccb)
![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/73cf3759-3b8c-4620-9e5b-3ff230c616dc)
![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/99a2c588-9ec6-4381-acf8-71fe78b1d026)
![#05](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/ce55363a-d246-426b-b20e-15650b85ddae)
![#06](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/6740a0cd-5b89-4971-9ce3-4763b74ac7e8)
![#07](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/dc701da8-8185-42e7-aea6-fde77441fe8b)

**Reply by Avel757** [reply]
Got it, when I’ll be home, I will make screenshots. Appreciate your time

**Reply by Avel757** [reply]
![image](https://github.com/aclap-dev/video-downloadhelper/assets/100200112/fefe0130-f46a-41ea-a3df-606235712323)

![image](https://github.com/aclap-dev/video-downloadhelper/assets/100200112/9fbe224b-c859-4fe2-b8d9-12b3d8a23904)
It says dowloading as usual

![image](https://github.com/aclap-dev/video-downloadhelper/assets/100200112/895fab3e-c349-4c91-9b6f-d18f63f51a30)

SideDownload error:
[tls @ 000001a543edad80] IO error: End of file
[https @ 000001a543edb6c0] Stream ends prematurely at 704138, should be 5125068
[tls @ 000001a543edad80] IO error: End of file
[aac_adtstoasc @ 000001a543f4fb00] Error parsing ADTS frame header!
[mp4 @ 000001a542508140] Error applying bitstream filters to an output packet for stream #1: Invalid data found when processing input
[aost#0:1/copy @ 000001a544de4c00] Error submitting a packet to the muxer: Invalid data found when processing input
[out#0/mp4 @ 000001a544d9f200] Error muxing a packet

![image](https://github.com/aclap-dev/video-downloadhelper/assets/100200112/627c81e9-e352-4c17-ab20-9dfcfbaab167)

**Reply by GrampaWildWilly** [reply]
Further to the questions below, I don't think this is anything to do with antivirus or firewall.  You are able to get other things with VDH, right?

On your VDH status screen, it shows 0% & what is allegedly the download speed.  But it is my understanding that if it doesn't also show an expected time until completion, then it has actually not started downloading.  It's still trying to figure out what to do before starting the download & the error actually occurs during the phase of VDH trying to figure out what it's got & how to deal with it.

It's too bad this is a subscription site.  I would want to look at the Network Monitor & see what's in there.  That is pretty much an improvisational exercise so I don't think I could even give you instructions for what I would try.  If there's older versions of VDH & CoApp that used to work for you on this site, I say you should go back to that setup & use it.  Turn off automatic updates so you don't accidentally get pushed back onto this version that doesn't work for you.

Or get a screen recorder (do some web searching on the subject) & record the content instead of download it.  That's a suboptimal solution but it's probably the best you're going to be able to do with this.

**Reply by GrampaWildWilly** [reply]
Maybe I'm not entirely correct when I say the download didn't start.  You did get a small file.  I think that proves neither antivirus nor firewall is interfering.  But something is & I don't know how to go about figuring out what since I can't get into the site.  Oh well.  It's not the first site VDH can't handle & it will probably not be the last.

**Comment by Mattmjs** [comment]
One thing about the vpn not mentioned, is it a browser extension or a desktop vpn. It would be a problem if using a browser extension.
And also have you checked it's not the anti-virus or firewall interfering.

**Reply by Avel757** [reply]
It s desktop vpn, how do i check firewall and anti-virus?

**Reply by Mattmjs** [reply]
Just check nothing is in quarantine but you could be ok on that and check nothing is blocked by firewall.
With the vpn, have you tried different protocols for the connection to see if this will make a difference.

**Reply by Mattmjs** [reply]
If you haven't already done so , go to the latest companion app as well. I'm thinking it could be the vpn.
The vpn I was using some time ago it would cause the connection stall down to nothing. So if it's doing this maybe the companion app thinks the connection has been lost therefore it ends the download.

You can monitor the connection in task manager to see what it is doing when downloading.
